### PR TITLE
dialects: (x86) PR5_3 - single operand instructions - source and destination

### PR DESCRIPTION
**Pull request by KGrykiel**
next part of PR https://github.com/xdslproject/xdsl/pull/2398. added the case where the single operand is both a source and destination + the not operation

### CODE REVIEW

File: xdsl/dialects/x86/ops.py [hunk 1]
```diff
@@ -320,6 +321,48 @@ class PushOp(ROperationSrc[GeneralRegisterType]):
...
+        super().__init__(
+            operands=[source],
+            attributes={
+                "comment": comment,
+            },
+            result_types=[destination],
+        )
+
+    def assembly_line_args(self) -> tuple[AssemblyInstructionArg | None, ...]:
+        return (self.destination,) if self.destination else (self.source,)
```
Comment (superlopuh) [inline xdsl/dialects/x86/ops.py:352]: Both of these are guaranteed not to be `None`, might as well just pick one

Comment (KGrykiel) [reply xdsl/dialects/x86/ops.py:352]: artifact from the previous version, my bad

**Comment by KGrykiel** [comment]
@superlopuh is this fine to be merged now?

**Review by superlopuh — APPROVED** [review]

**Comment by KGrykiel** [comment]
God, I hate merge conflicts, I'll just start over...

**Pull request closed without merging**